### PR TITLE
feat: add customization on redis db

### PR DESCRIPTION
**Pull request by zyyw**

### CODE REVIEW

File: values.yaml [hunk 1]
```diff
@@ -814,6 +814,16 @@ redis:
...
     priorityClassName:
+    # # jobserviceDatabaseIndex defaults to "1"
+    # # registryDatabaseIndex defaults to "2"
+    # # trivyAdapterIndex defaults to "5"
+    # # harborDatabaseIndex defaults to "0", but it can be configured to "6"
+    # # cacheLayerDatabaseIndex defaults to "0", but it can be configured to "7"
+    jobserviceDatabaseIndex: "1"
+    registryDatabaseIndex: "2"
+    trivyAdapterIndex: "5"
+    # harborDatabaseIndex: "6"
```
Comment (MinerYang) [inline values.yaml:825]: is it better to add a comment clarify it is optional for  `harborDatabaseIndex` and `cacheLayerDatabaseIndex` ?

File: templates/_helpers.tpl [hunk 2]
```diff
@@ -164,30 +164,46 @@ app: "{{ template "harbor.name" . }}"
...
-    {{- $index := ternary "5" .external.trivyAdapterIndex (eq .type "internal") }}
+    {{- $index := ternary .internal.trivyAdapterIndex .external.trivyAdapterIndex (eq .type "internal") }}
+    {{- printf "%s/%s?idle_timeout_seconds=30" (include "harbor.redis.url" $) $index -}}
+  {{- end }}
+{{- end -}}
+
+/*scheme://[:password@]addr/db_index?idle_timeout_seconds=30*/
+{{- define "harbor.redis.urlForHarbor" -}}
+  {{- with .Values.redis }}
+    {{- $index := ternary .internal.harborDatabaseIndex .external.harborDatabaseIndex (eq .type "internal") }}
```
Comment (MinerYang) [inline templates/_helpers.tpl:191]: either `internal.harborDatabaseIndex` OR `external.harborDatabaseIndex` ?

**Review by chlins — APPROVED** [review]
lgtm

File: values.yaml (hunk 1 above)
Comment (zyyw) [reply values.yaml:825]: add the comment in line 820 & 821

**Pull request merged**